corehq/apps/hqwebapp/static/hqwebapp/js/knockout_bindings.ko.js

```diff
@@ -171,6 +171,174 @@ hqDefine("hqwebapp/js/knockout_bindings.ko", ['jquery', 'knockout', 'jquery-ui/u
+                    } else {
+                        lastClickedRow = $(this).parent().children().eq(0);
+                    }
+
+                    var firstRow = null,
+                        start = null,
+                        end = null;
+                    if (shiftSelectedIndex < lastClickedIndex) {
+                        start = shiftSelectedIndex;
+                        end = lastClickedIndex;
+                        firstRow = shiftClickedRow;
+                    } else {
+                        start = lastClickedIndex;
+                        end = shiftSelectedIndex;
+                        firstRow = lastClickedRow;
+                    }
+
+                    var next = firstRow;
+                    for (var i = start; i <= end; i++) {
+                        next.addClass('selected-for-sort');
+                        next = next.next();
+                    }
+                } else {
+                    $(this).addClass("selected-for-sort").addClass('last-clicked')
+                           .siblings().removeClass('selected-for-sort').removeClass('last-clicked');
+                }
+            });
+
+            $(element).on('click', '.send-to-top', function (e) {
+                $(this).parent().parent().addClass("moving").siblings().removeClass('moving');
+
+                // update UI
+                var row = $(this).parent().parent();
+                var currentIndex = row[0].attributes['data-order'].value;
+                row.parent().prepend(row);
+
+                // update KO
+                list().unshift(list().splice(currentIndex, 1)[0]);
+            });
+
+            $(element).on('click', '.send-to-bottom', function (e) {
+                $(this).parent().parent().addClass("moving").siblings().removeClass('moving');
+
+                var row = $(this).parent().parent();
+                var currentIndex = row[0].attributes['data-order'].value;
+
+                var lastSelectedRowIndex = null;
+                for (var i = 0; i < list().length; i++) {
+                    if (list()[i].selected()) {
+                        lastSelectedRowIndex = i;
+                    }
+                }
+
+                if (currentIndex < lastSelectedRowIndex) {
+                    // Update UI
+                    $('.isSelectedForExport').addClass('selected-for-export-above-last');
+                    row.removeClass('selected-for-export-above-last');
+                    $('.selected-for-export-above-last:nth-child('
+                        + ($('.selected-for-export-above-last').length + 1) + ')').after(row);
+                    $('.selected-for-export-above-last').removeClass('selected-for-export-above-last');
+
+                    // Update KO
+                    var currentListItem = list().splice(currentIndex, 1)[0];
+                    list().splice(lastSelectedRowIndex, 0, currentListItem);
```

Comment:
Does this work? I would think that `list.splice` would update the value stored in the observable, but `list().splice` would return the value, and then update the returned value instead of the observable itself. I haven't tested this, so I'm not sure.

Line 154 uses the first construct (`list.splice`) to update the value. If they're equivalent, then I think this is fine.

Reply:
Thanks for pointing that out.  I updated the send to top/bottom KO code, and that also allowed me to get rid of extra jQuery code.

I tried using the observable itself in the sortable code, but ran into some issues.  This section mixes jQuery/KO and they don't play nicely.  From testing, it seems that updating ```list()``` updates the underlying data without notifying the UI, whereas updating ```list``` notifies the UI.

Reply:
Planning on returning to this part at a later time